src/App.tsx

```diff
@@ -1,40 +1,65 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect } from 'react';
 import {
   BrowserRouter as Router,
   Route,
   Switch,
   RouteComponentProps,
 } from 'react-router-dom';
 import 'hds-core';
+import querystring, { ParsedUrlQuery } from 'querystring';
+import {
+  Tokens,
+  setTokens,
+  getTokens,
+  convertParamsToTokens,
+  isValidAuthParams,
+  AuthContext,
+} from './auth/auth-context';
 import Main from './components/main/Main';
 import NavigationAndFooterWrapper from './components/navigation-and-footer-wrapper/NavigationAndFooterWrapper';
 import './App.scss';
 import TargetPage from './target/page/TargetPage';
 
 export default function App(): JSX.Element {
+  const existingTokens: Tokens | undefined = getTokens();
```

Comment:
I wonder if the term _Token_ could be unnecessarily abstract for us? Would it be more readable with a prefix _auth_ in it to indicate what those tokens are? In other words, tokens obtained from authentication. :thinking:

Reply:
Do you mean the name of the type and variables? I'll test how does it look

Reply:
Replaced tokens with authTokens